DeviceCommunications/DeviceComsProtocolParser.cs

```diff
@@ -0,0 +1,194 @@
+﻿using DeviceCommunications.Interfaces;
+using EEGDataHandling;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DeviceCommunications
+{
+    public class DeviceComsProtocolParser : IDeviceComs, IEEGData
+    {
+        // Set capacity to 30 seconds at 500 samples/second
+        private const int QUEUE_CAPACITY = 500 * 30;
+
+        // use an abstracted device coms stream.
+        IDeviceComsStream _DeviceComsStream;
+
+        private object _dataPointsLock = new object();
+
+        private readonly ConcurrentQueue<(long, EEGData)> _dataPoints;
+
+        public DeviceComsProtocolParser(IDeviceComsStream deviceComsStream)
+        {
+            _DeviceComsStream = deviceComsStream;
+
+            if (_DeviceComsStream.Connect())
+            {
+                // if device is connected then subscribe to it's DataReceived events
+                _DeviceComsStream.DataReceivedEvent += DataReceivedEventHandler;
+            }
+
+            _dataPoints = new ConcurrentQueue<(long, EEGData)>();
+        }
+
+        public event EventHandler DataUpdated;
+
+        void IDeviceComs.SendGetDataSinceLastRequest()
```

Comment:
I'm curious about implementing the interfaces explicitly here - as far as I know, that's only strictly necessary if you have a method signature collision between the two interfaces and want to define different logic for each. Are there other benefits to doing this?

Reply:
Visual studio asked if I wanted to do it and I liked the idea of knowing immediately where they came from, no technical reason.